src/parser.rs

```diff
@@ -386,8 +408,21 @@ impl Parser {
         let reader = fs::File::open(&parser_path)
             .with_context(|_| format_err!("Error when opening the parser file"))?;
 
-        Ok(from_read(reader)
-            .with_context(|_| format_err!("Error when deserializing the parser"))?)
+        let mut parser: Parser = from_read(reader)
+            .with_context(|_| format_err!("Error when deserializing the parser"))?;
+
+        if let Some(info) = config.license_info {
+            let license_path = folder_name.as_ref().join(&info.filename);
+            let license_content = fs::read_to_string(&license_path)
+                .with_context(|_| format_err!("Error when reading the license"))?;
+
+            parser.license_info = Some(LicenseInfo {
+                content: license_content,
+                filename: info.filename,
+            })
+        };
+
+        Ok(parser)
```

Comment:
You don't have to set the `license_info` as it is already deserialized few lines before.